kasa/utils.py

```diff
@@ -0,0 +1,13 @@
+"""Python library supporting TP-Link Smart Home devices."""
+
+from __future__ import annotations
+
+
+def merge(dest: dict, source: dict) -> dict:
+    """Update dict recursively."""
+    for k, v in source.items():
+        if k in dest and isinstance(v, dict):
```

Comment:
This was `Mapping` before, but mypy complained about it so I changed it to `dict`.. need to double check this